src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,379 @@
+  // Return all entries in `src/content/docs/en/`
+  return id.startsWith('en/');
+});
+---
+```
+
+### `getEntry()`
+
+`getEntry()` is function that returns a specific entry in a collection by entry ID (file path relative to the collection). Both of these are required parameters.
+
+```astro
+---
+import { getEntry } from 'astro:content';
+
+const enterprise = await getEntry('blog', 'enterprise.md');
+---
+```
+
+### Data returned from a collection query
+
+`getCollection()` and `getEntry()` will return entries that include:
+ - `id` - a unique ID using the file path relative to `src/content/[collection]`
+ - `slug` - a URL-ready slug. Defaults to the ID without the file extension.
+ - `data` - an object of frontmatter properties inferred from your collection schema. Defaults to `any` if no schema is configured.
+ - `body` - a string containing the raw, uncompiled body of the Markdown or MDX document.
+ - `render()` - a function that returns the compiled body of the Markdown or MDX document via the `<Content />` component ([See complete documentation](#render)).
+
+Querying your content files with `getCollection()`or `getEntry()` allows you to use frontmatter properties from an entry's `data` object in [JSX-like expressions](/en/core-concepts/astro-components/#jsx-like-expressions) or pass props to other components, such as a layout. You can optionally add type safety with a built-in utility.
+
+For example, you can use a `getCollection()` query to filter and then display a list of links to all your published blog posts:
+```astro
+---
+// src/pages/index.astro
+import { getCollection } from 'astro:content';
+
+// Get all published blog posts
+const blogPosts = await getCollection('blog', ({ data }) => {
+  return data.status === 'published';
+});
+---
+<ul>
+  {allBlogPosts.map(post => (
+    <li>
+      <a href={post.data.slug}>{post.data.title}</a>
+      <time datetime={post.data.publishedDate.toISOString()}>
+        {post.data.publishedDate.toDateString()}
+      </time>
+    </li>
+  ))}
+</ul>
+```
+
+### Collection entry types
+
+If a page or component uses content from a `getCollection()` or `getEntry()` query, you can use the `CollectionEntry` utility to type its props:
+
+```astro /CollectionEntry([<.+>])?/
+---
+// src/components/BlogCard.astro
+import type { CollectionEntry } from 'astro:content';
+
+interface Props {
+  // Get type of a `blog` collection entry
+  post: CollectionEntry<'blog'>;
+}
+
+// `post.data` will match your collection schema
+const { post } = Astro.props;
+---
+```
+
+## Rendering page content from entries
+
+To render the content of your Markdown and MDX entries, use the `<Content />` component returned by the `render()` function. This allows you to generate pages from your content entries (see [Generating pages from content collections](#generating-pages-from-content-collections)), add post previews to your homepage, or display your content elsewhere on your site.
```

Comment:
``````suggestion
Use the provided `render()` function to generate pages from your content entries (see [Generating pages from content collections](#generating-pages-from-content-collections)), add post previews to your homepage, or display your content elsewhere on your site. With this function, you can access an entry's Markdown content, headings and any injected frontmatter.

```
const { Content, headings, injectedFrontmatter } = entry.render();
```
``````

@delucis Were you thinking of something like this? If so, I might suggest we change the headings to look maybe like:

## Rendering entries (with `render()`??)
### Accessing content from `render()`
### Accessing headings from `render()`
### Accessing injected frontmatter from `render()`

(Proposing those perhaps less-than-ideal headings for now because I think they'll distinguish themselves enough as Algolia search terms. We can think about better headings as we prepare for this to come out of experimental. But, at least it's very clear right now.

Reply:
Yes, exactly, something like that 👍